src/main/java/org/folio/processing/mapping/mapper/writer/marc/MarcRecordModifier.java

```diff
@@ -102,18 +106,9 @@ public void modifyRecord(List<MarcMappingDetail> mappingDetails) {
   }
 
   public DataImportEventPayload getResult(DataImportEventPayload eventPayload) {
-    try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
-      MarcWriter streamWriter = new MarcStreamWriter(new ByteArrayOutputStream());
-      MarcJsonWriter jsonWriter = new MarcJsonWriter(os);
-      streamWriter.write(marcRecordToChange);
-      jsonWriter.write(marcRecordToChange);
-      recordToChange.getParsedRecord().setContent(new JsonObject(new String(os.toByteArray())).encode());
-      String resultKey = marcMappingOption == MODIFY ? marcType.value() : getMatchedMarcKey();
-      eventPayload.getContext().put(resultKey, Json.encode(recordToChange));
-    } catch (Exception e) {
-      LOGGER.error("Can not put the modified record to the event payload", e);
-      throw new IllegalStateException(e);
-    }
+    recordToChange.getParsedRecord().setContent(mapRecordRepresentationToJsonString(marcRecordToChange));
```

Comment:
I am not able to discern what `recordToChange` is. it is a field in this class? Thinking if there is a way to make it obvious. Maybe calling a getter instead.

Reply:
yes correct it is the class field, changed the field usage to more explicit using `this` keyword